packages/babel-plugin-transform-remove-console/src/index.js

```diff
@@ -1,14 +1,53 @@
 "use strict";
 
-module.exports = function() {
+module.exports = function({ types: t }) {
   return {
     name: "transform-remove-console",
     visitor: {
       CallExpression(path) {
-        if (path.get("callee").matchesPattern("console", true)) {
-          path.remove();
+        const callee = path.get("callee");
+
+        if (!callee.isMemberExpression()) return;
+
+        if (isConsole(callee)) {
+          // console.log()
+          if (path.parentPath.isExpressionStatement()) {
+            path.remove();
+          } else {
+            path.replaceWith(createVoid0());
+          }
+        } else if (isConsoleBind(callee)) {
+          // console.log.bind()
+          path.replaceWith(createNoop());
         }
       },
+      MemberExpression: {
+        exit(path) {
+          if (!isConsole(path)) return;
+          if (!path.parentPath.isMemberExpression()) {
+            path.replaceWith(createNoop());
+          }
+        }
+      }
     },
   };
+
+  function isConsole(memberExpr) {
```

Comment:
And the test needs to check both `a = console.log.call` and `console.log.call()`